tests/cases/fourslash/codeFixMissingTypeAnnotationOnExports36-conditional-releative.ts

```diff
@@ -0,0 +1,31 @@
+/// <reference path='fourslash.ts'/>
+
+// @isolatedDeclarations: true
+// @declaration: true
+
+// @Filename: /code.ts
+////const A = "A"
+////const B = "B"
+////export const AB = Math.random()? A: B;
+verify.codeFixAvailable([
+    {
+      "description": "Add annotation of type '\"A\" | \"B\"'"
+    },
+    {
+      "description": "Add annotation of type 'typeof A | typeof B'"
+    },
+    {
+      "description": "Add satisfies and an inline type assertion with '\"A\" | \"B\"'"
+    },
+    {
+      "description": "Add satisfies and an inline type assertion with 'typeof A | typeof B'"
+    }
+])
+verify.codeFix({
+    description: "Add satisfies and an inline type assertion with 'typeof A | typeof B'" ,
+    index: 3,
+    newFileContent:
+`const A = "A"
+const B = "B"
+export const AB = (Math.random() ? A : B) satisfies typeof A | typeof B as typeof A | typeof B;`
```

Comment:
Actually [it changes the type](https://www.typescriptlang.org/play?ts=5.4.5#code/MYewdgzgLgBAgjAvDARHFAoUlYCEmq6YCmAHgA4gBOs208+yAFALICGUAFgHRVtgATEAFsmAShgB+eDABcMXBIgcAlhABmK4hBhQAnuWIh1MgD66DRk-jY79h42YsPrAbgxlKNGHVhxcAEwErBw8fIIi4lIy8oquQA), I guess this will break downstream code.

Reply:
Yeah, which can be fine, though maybe that means the fix needs to offer both unwidened and widened types if they differ.

Reply:
Done. Although I have two issues that I need to note.

1. I needed to expose `TypeChecker.getWidenedLiteralType` as an internal API to imitate what the type checker is doing during d.ts emit.
2. In the code fixer it's querying for the type of the same node twice, and it seems to affect the freshness to that manner, and results in `isFreshLiteralType` returning `False` . For the reasons that I don't understand, `getWidenedLiteralType` is conditionally widening a literal type only if it's fresh. So the  updated test file 36-conditional-relative.ts  an inline annotation with `string` is not being added. Would you know what's the best approach to take here?